### ISSUE TITLE
linter: react/display-name false positive for directly invoked JSX callback returned by a factory

### ISSUE THREAD

**Issue by ericrafalovsky** [oxc-project/oxc#25165]
### What version of Oxlint are you using?

1.7.6

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "parser": {
    "extension": "tsx",
    "allowReturnOutsideFunction": true,
    "preserveParens": true,
    "allowV8Intrinsics": true,
    "semanticErrors": true
  },
  "linter": {
    "config": "{\"categories\":{\"correctness\":\"off\"},\"rules\":{\"react/display-name\":\"error\"},\"plugins\":[\"eslint\",\"oxc\",\"typescript\",\"unicorn\",\"vue\",\"react\"]}"
  }, 
}
```

### What happened?

`react/display-name` reports a false positive for directly invoked JSX callback returned by a factory

```tsx
const createCallback = () => () => <div />;

createCallback()();
```

See playground [link](https://playground.oxc.rs/?lintRules=react%2Fdisplay-name&options=%7B%22run%22%3A%7B%22lint%22%3Atrue%2C%22formatter%22%3Afalse%2C%22transform%22%3Afalse%2C%22isolatedDeclarations%22%3Afalse%2C%22whitespace%22%3Afalse%2C%22mangle%22%3Afalse%2C%22compress%22%3Afalse%2C%22scope%22%3Atrue%2C%22symbol%22%3Atrue%2C%22cfg%22%3Afalse%7D%2C%22parser%22%3A%7B%22extension%22%3A%22tsx%22%2C%22allowReturnOutsideFunction%22%3Atrue%2C%22preserveParens%22%3Atrue%2C%22allowV8Intrinsics%22%3Atrue%2C%22semanticErrors%22%3Atrue%7D%2C%22linter%22%3A%7B%22config%22%3A%22%7B%5C%22categories%5C%22%3A%7B%5C%22correctness%5C%22%3A%5C%22off%5C%22%7D%2C%5C%22rules%5C%22%3A%7B%5C%22react%2Fdisplay-name%5C%22%3A%5C%22error%5C%22%7D%2C%5C%22plugins%5C%22%3A%5B%5C%22eslint%5C%22%2C%5C%22oxc%5C%22%2C%5C%22typescript%5C%22%2C%5C%22unicorn%5C%22%2C%5C%22vue%5C%22%2C%5C%22react%5C%22%5D%7D%22%7D%2C%22transformer%22%3A%7B%22target%22%3A%22es2015%22%2C%22useDefineForClassFields%22%3Atrue%2C%22experimentalDecorators%22%3Atrue%2C%22emitDecoratorMetadata%22%3Atrue%2C%22optimizeEnums%22%3Atrue%2C%22optimizeConstEnums%22%3Atrue%7D%7D&code=const+createCallback+%3D+%28%29+%3D%3E+%28%29+%3D%3E+%3Cdiv+%2F%3E%3B%0A%0AcreateCallback%28%29%28%29%3B)

**Comment by camc314** [comment]
Hi! Thanks for reporting this issue.

The upstream `eslint-plugin-react` also reports the same diagnostic. Immediate invocation does not change how the anonymous JSX-returning funciton is classified.

For this specific case, i'd suggest disabling the rule locally:

```tsx
// oxlint-disable-next-line react/display-name -- callback, not a component
const createCallback = () => () => <div />;
```

Alternatively, give the returned function a name:

```tsx
const createCallback = () => function Callback() {
  return <div />;
};
```
